src/App.jsx

```diff
@@ -139,6 +138,7 @@ function App() {
     for(let i in Object.keys(soundStates)){
       soundStates[Object.keys(soundStates)[i]].isPlaying = true
     }
+    localStorage.setItem("zenmix-state", JSON.stringify(soundStates))
```

Comment:
```suggestion
    setSoundStates((prevStates) => {
      const updatedStates = { ...prevStates };
      Object.keys(updatedStates).forEach((key) => {
        updatedStates[key].isPlaying = true;
      });
      localStorage.setItem("zenmix-state", JSON.stringify(updatedStates));
      return updatedStates;
    });
```

This way it creates a shallow copy of the `prevStates` object and assigns it to `updatedStates`.
This ensures that the original `prevStates` object remains unmodified while allowing updates to be made to the new `updatedStates` object